### DISCUSSION TITLE
Install older file version

### DISCUSSION THREAD

**Discussion by StefanKoell** [wixtoolset/issues#6930] [Questions]
Hi!

I currently have an issue where a .dll file disappears after upgrade. After some trial and error I found out that one of the files I want to deploy had a version 42.42.42.42424 but now the MSI I built for the upgrade has a version 7.0.something. As a result, after upgrading with the MSI, the dll file is gone. It has been removed from the previous MSI during uninstall but for some reason it hasn't been installed when the new MSI unpacks the files.

Uninstalling/installing manually works and the file is deployed.

Any idea how I can instruct the MSI or configure that file to always deploy - regardless if it's a newer or older version?

Thanks,
Stefan

**Comment by glen-nicol** [comment]
This is Windows Installer behavior. I am not sure WIX can do much to affect it directly. See following docs:

* https://learn.microsoft.com/en-us/windows/win32/msi/file-versioning-rules
* https://learn.microsoft.com/en-us/windows/win32/msi/both-files-have-a-version

This bit me once. Luckily we had control over the file version so we just revved that to be larger. For files that you don't have control over it is really gonna depend on your exact scenario. It looks like there are [options to control the behavior to some extent](https://learn.microsoft.com/en-us/windows/win32/msi/reinstallmode). 

Alternatively, since windows installer operates on components. You could group this file with some other component to avoid the comparison (and link it to whatever file versioning is used for the other component you group it with). If you're using Heat to harvest files I am not sure of a way to accomplish this in an easy way. Might need to have some custom code or XML transforms.

Good Luck!
